service/remote_processor_service.py

```diff
@@ -90,19 +90,42 @@ def ProcessResponse(self, request: ProcessResponseRequest, context):
             context.set_code(grpc.StatusCode.NOT_FOUND)
             context.set_details(f'Processor {processor_name} does not exist!')
             raise KeyError(f"Processor {processor_name} does not exist! Check your config file {self._config_file}")
-        
+
     def start(self):
         """Start the server on port 2796 (ARYN on a keypad)
 
         Returns:
             Server: a grpc server object
         """
         logging.info(PAPRIKA_ASCII_ART)
+        logging.info("Starting service in insecure mode")
         server = grpc.server(futures.ThreadPoolExecutor(max_workers=TP_MAX_WORKERS))
         add_RemoteProcessorServiceServicer_to_server(self, server)
         server.add_insecure_port("[::]:2796")
         server.start()
         logging.info("RPS started on port 2796")
         return server
-        
-    
\ No newline at end of file
+
+    def start_secure(self, certfile, keyfile):
+        """Start the server on port 2796 (ARYN on a keyboard) with encryption
+
+        Returns:
+            Server: a grpc server object
+        """
+        logging.info(PAPRIKA_ASCII_ART)
+        logging.info("Starting service in secure mode")
+        server = grpc.server(futures.ThreadPoolExecutor(max_workers=TP_MAX_WORKERS))
+        add_RemoteProcessorServiceServicer_to_server(self, server)
+        with open(keyfile, 'rb') as f:
+            private_key = f.read()
+        with open(certfile, 'rb') as f:
+            cert_chain = f.read()
+        channel_credentials = grpc.ssl_server_credentials(
+            private_key_certificate_chain_pairs=[(private_key, cert_chain)],
+            root_certificates=None,
+            require_client_auth=False
+        )
+        server.add_secure_port("[::]:2796", channel_credentials)
```

Comment:
This is the only different code.  Rather than two whole functions, can we just key off of the passed-in parameters, like if certfile is None?